File: beginner/java/07_reverse_string/Solution.java
/**
 * Challenge 07: Reverse String
 * TODO: Implement the reverseString method
 */
public class Solution {
    
    /**
     * Reverse a string
     * 
     * @param text A string to reverse
     * @return The reversed string
     */
public static String reverseString(String text) {
    if (text == null || text.isEmpty() || text.length() == 1) {
        return text;
    }
    StringBuilder result = new StringBuilder();
    for (int i = text.length() - 1; i >= 0 ; i--) {
        result.append(text.charAt(i));
    }
    return result.toString();
}

    
    public static void main(String[] args) {
        // Test your method
        System.out.println("reverseString(\"hello\") = " + reverseString("hello"));
        System.out.println("reverseString(\"Java\") = " + reverseString("Java"));
        System.out.println("reverseString(\"\") = \"" + reverseString("") + "\"");
    }
}

File: beginner/python/07_reverse_string/solution.py
"""
Challenge 07: Reverse String
TODO: Implement the reverse_string function
"""

def reverse_string(text: str) -> str:
    """
    Reverse a string.
    
    Args:
        text: A string to reverse
    
    Returns:
        str: The reversed string
    """
    return reverse_string_helper(text)

def reverse_string_helper(text): 
    if text == "":
        return text
    reversed_text = ""
    for char in text:
        reversed_text = char + reversed_text
    return reversed_text
        
        
        
if __name__ == "__main__":
    # Test your function
    print(f"reverse_string('hello') = {reverse_string('hello')}")
    print(f"reverse_string('Python') = {reverse_string('Python')}")
    print(f"reverse_string('') = '{reverse_string('')}'")
